### ISSUE TITLE
Proposal to support organization-level shipping information more extensively

### ISSUE THREAD

**Issue by pascalfleury** [schemaorg/schemaorg#3617]
This is a proposal from Google to support a richer representation of shipping details (such as delivery cost and speed) and make this kind of data explicit. If adopted by schema.org and publishers, we consider it likely that search experiences and other consuming systems could be improved by making use of such markup.

This change introduces a new type, `ShippingService`, that groups shipping constraints (delivery locations, time, weight and size limits and shipping rate). Redundant fields from `ShippingRateSettings` are therefore been deprecated in this proposal.

As a consequence, the following changes are also proposed:

- some fields in `OfferShippingDetails` have moved to `ShippingService`;
- `ShippingRateSettings` has more ways to specify the shipping rate, proportional to the order price or shipping weight;
-  linking from the `Offer` should now be done with standard Semantic Web URI linking.
 
Given the size of the proposal, an external [Google Docs](https://docs.google.com/document/d/1kTJJKnnWtnoJuUK0cmtDdPH3JLZILyfN6IXICXti4wQ/edit?usp=sharing&resourcekey=0-BernNphrZhVov9lX7u2aNg) document is created for review and inline comments.

**Comment by pascalfleury** [comment]
The new structure can be navigated at: https://1-dot-schemadotorg1.ew.r.appspot.com/ShippingService

**Comment by ilanadavis** [comment]
Every time I see `ShippingService` I think you're asking for the service provider (e.g. USPS, UPS, FedEx...), not the conditions of shipping the product.

**Comment by alex-jansen** [comment]
Adding @jvdriel and @mfhepp to also take a look.

**Comment by pascalfleury** [comment]
@ilanadavis In a way this is it though. A merchant could point their offer-level `shippingDetails` to the `ShippingService` from eg. DHL or FedEx. DHL or FedEx may have multiple such `ShippingService` and would be a shipping service provider. 

Do you suggest to change the name to something more detailed like `ShippingServiceConditions`, of which a service provider like DHL may possibly have several ?

**Comment by ilanadavis** [comment]
To be clear, I fully support your changes and look forward to them. I'm not entirely sure how merchants will get the most up-to-date data since it changes frequently, but I love the idea. 

> Do you suggest to change the name to something more detailed like ShippingServiceConditions, of which a service provider like DHL may possibly have several ?

@pascalfleury Yes, I think that makes so much more sense. Lots of merchants ship with different providers depending on the product and may have different conditions at the product and service provider level. 

Are you expecting them to also include the service provider name or is that not important? That wasn't clear to me in your examples.

**Comment by Tiggerito** [comment]
Like with returns policies, it would be nice to clarify how more specific (Product level over Organization level, Variant level over Product level) shipping information overrides the more generic information.

e.g. a specific product or group of products may have their own shipping rules.

**Comment by pascalfleury** [comment]
@ilanadavis: I can change it to `ShippingConditions` (it's a bit shorter than my previous suggestion and just as clear I would claim). The examples indeed did not illustrate this, though each `ShippingConditions` object can have a `name`, `description` and all the other properties from the type `Thing`. Also, a particular shipping service provider like DHL can have their `ShippingConditions` grouped under their `Organization` so that the data about the service provider can also be as extensive as needed.

**Comment by pascalfleury** [comment]
@Tiggerito: I think it should be a Type-wide wholesale override. If the field contains an `@id` link to a `ShippingConditions`, that will be used, and no other fields should be present [*]. If at the product-level or variant-level the shipping conditions are different, it can be detailed right there with all the necessary fields. That means the to-be used `ShippingConditions` is explicit in both cases: as a link or as an in-place object. 

[*] I just tried on json-ld.org/playground and it seems to accept things like this:
```
"shippingService": {
        "@type": "ShippingService",
        "@id": "https://www.example.com/shipping.html#us_and_ca",
        "doesNotShip": "True"    // <<<== this is ill-defined IMO
      }
```

**Comment by Tiggerito** [comment]
@pascalfleury I re-read the document and what you said makes sense. The Organization is a place where shared ShippingConditions can be stored. But the ShippingDetails is always at the ProductGroup or Product level.

This is how I currently deal with Shipping Details:

In the back end the owner can define a global set of shipping details. Each contains the fields Google currently support, like location and times, but not specifics about dimensions. Each entry also has conditions for what product the entry can apply to. This can include a price range and a weight range.

When I'm generating the structured data for a page I include the entries where the product matches the conditions.

This change looks like it will let me change from filtering out the conditions on the server, to including them in the Structured Data on the product page.

Then the consumers of the data can calculate which ShippingConditions are a match and therefore what rates are available when ordering a specific number of the product. Currently, you can only provide prices for shipping one.

The split also means it's easier to provide product specific information as well as shared shipping information without the need for repetition.

Your example in the document at the end for using Organization. It looks like you are referencing ShippingConditions for a product that are on a shipping page. This cross-referencing between pages could greatly reduce the bloat this has on the product page, if supported by Google.

**Comment by pascalfleury** [comment]
@Tiggerito 

>The Organization is a place where shared ShippingConditions can be stored. But the ShippingDetails is always at the ProductGroup or Product level.

Indeed, and this is already the case. This change also separates the two meanings of eg. `width`, `height`, `weight` as description of the product (in `ShippingDetails`) and as constraints in the `ShippingConditions` where they can be expressed as a range (`QuantitativeValue` has min and max). 
 
> In the back end the owner can define a global set of shipping details. Each contains the fields Google currently support, like location and times, but not specifics about dimensions. Each entry also has conditions for what product the entry can apply to. This can include a price range and a weight range.
> 
> When I'm generating the structured data for a page I include the entries where the product matches the conditions.
> 
> This change looks like it will let me change from filtering out the conditions on the server, to including them in the Structured Data on the product page.
> 
> Then the consumers of the data can calculate which ShippingConditions are a match and therefore what rates are available when ordering a specific number of the product. Currently, you can only provide prices for shipping one.

Some shipping constraints are not available at the time the product is listed or even rendered on a page (eg. shipping destination, number of items, wanted delivery speed or customer tier if the user is not logged in). The `ShippingDetails` attached to a product should contain information about the product itself only, the rest gets moved to the new `ShippingConditions` in this proposal.
Note that schema.org does not specify a cardinality, so that we could specify multiple `ShippingConditions` links so that the appropriate one gets selected at the consumer side.
 
> The split also means it's easier to provide product specific information as well as shared shipping information without the need for repetition.
> 
> Your example in the document at the end for using Organization. It looks like you are referencing ShippingConditions for a product that are on a shipping page. This cross-referencing between pages could greatly reduce the bloat this has on the product page, if supported by Google.

Indeed. This is where we are trying to get at.

**Comment by Tiggerito** [comment]
@pascalfleury, thanks for the detailed answers.

**Comment by fthobe** [comment]
Has been merged and can be closed I believe. @pascalfleury

**Comment by alex-jansen** [comment]
Agreed, closing.

**Comment by jac-frc** [comment]
What about if the Schema is to describe that Shipping is free if the total order amount is over or equal to some currency value?

Update: Found it myself, use `ShippingRateSettings` https://schema.org/ShippingRateSettings
